compilers/concrete-compiler/compiler/lib/Common/Keysets.cpp

```diff
@@ -417,5 +418,98 @@ KeysetCache::getKeyset(const Message<concreteprotocol::KeysetInfo> &keysetInfo,
   return std::move(keyset);
 }
 
+Message<concreteprotocol::KeysetInfo> generate_generic_keyset_info(
+    std::vector<concrete_optimizer::utils::PartitionDefinition> partitionDefs,
+    bool generateFks) {
+  auto output = Message<concreteprotocol::KeysetInfo>{};
+  rust::Vec<concrete_optimizer::utils::PartitionDefinition> rustPartitionDefs{};
+  for (auto def : partitionDefs) {
+    rustPartitionDefs.push_back(def);
+  }
+  auto parameters = concrete_optimizer::utils::generate_generic_keyset_info(
+      rustPartitionDefs, generateFks);
+
+  auto skLen = (int)parameters.secret_keys.size();
+  auto skBuilder = output.asBuilder().initLweSecretKeys(skLen);
+  for (int i = 0; i < skLen; i++) {
```

Comment:
`for (auto sk : llvm::enumerate(parameters.secret_keys))` ?

Reply:
Didn't know this one. Pretty good, thanks !